cpython/search_pypi_top.py

```diff
@@ -138,39 +134,76 @@ def is_binary_file(args, fp):
         return True
     return False
 
+
 def grep(args, archive_filename, regex):
     for filename, fp in decompress(args, archive_filename):
         if is_binary_file(args, fp):
-            logging.info(f"ignore binary file: {archive_filename}: {filename}")
+            logger.info(f"ignore binary file: {archive_filename}: {filename}")
             continue
 
-        matchs = []
+        matches = []
         ignore = False
         lineno = 1
         # Split at Unix newline b'\n' byte
         for line in fp:
             if lineno == 1 and IGNORE_CYTHON and CYTHON_REGEX.match(line):
-                logging.info(f"ignore Cython file: {archive_filename}: {filename}")
+                logger.info(f"ignore Cython file: {archive_filename}: {filename}")
                 ignore = True
                 break
             if regex.search(line):
-                matchs.append((filename, line))
+                matches.append((filename, line))
             lineno += 1
 
-        if matchs and not ignore:
-            yield from matchs
+        if matches and not ignore:
+            yield from matches
+
+
+def search_file(filename, index, len_filenames, args, pypi_dir, regex):
+    lines = 0
+    results = []
+    filename = os.path.join(pypi_dir, filename)
+    percent = index * 100 / len_filenames
+    logger.warning(f"grep {filename} ({percent:.0f}%, {index}/{len_filenames})")
+
+    for name, line in grep(args, filename, regex):
+        line = line.decode('utf8', 'replace').strip()
+
+        result = f"{filename}: {name}: {line}"
+        print(result, flush=True)
```

Comment:
The print() operation is not atomic because of the newline. Maybe add manually "\n" and use end='', or send the results into a queue, and add a thread responsible to manage stdout to avoid any risk of corrupted output.

Reply:
Done via first option.